### PR TITLE
Documentation: Update first half of DG use cases

### PR DESCRIPTION
**Pull request by mounilsankar**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -735,28 +739,105 @@ Priorities: High (Must-Have) - * * *, Medium (Good-To-Have) - * *, Low (To-Forgo
...
+Use case ends.
+
+**Extensions**
+
+* 1a. MediConnect detects that the given command is invalid.
+    * 1a1. MediConnect shows an error message.
+    * 1a2. MediConnect requests the correct input
+
+      Use case resumes at step 1.
+* 1b. MediConnect detects that the given patient does not exist in the system.
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:776]: is this supposed to be index is invalid?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -852,8 +937,9 @@ Priorities: High (Must-Have) - * * *, Medium (Good-To-Have) - * *, Low (To-Forgo
 
 **MSS**
 
-1. User requests to view a patient’s upcoming appointments
-2. MediConnect shows the patient’s appointment list
+1. User requests to view a patient’s upcoming appointments.
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:940]: maybe `list` instead of `view` is better to align it with the command

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (mounilsankar) [reply docs/DeveloperGuide.md:776]: i think this is fine? Index invalid would be more specific?

Comment (yanghengtang) [reply docs/DeveloperGuide.md:776]: would that be too specific and define the design? hmm

Comment (derekjxtan) [reply docs/DeveloperGuide.md:776]: ok

**Review by derekjxtan — APPROVED** [review]
lgtm

**Pull request merged**